File: sdks/ts/src/sdk/models/shared/storagepolicymini.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum StoragePolicyMiniTypeEnum {
    StoragePolicy = "storage_policy"
}


// StoragePolicyMini
/** 
 * A mini description of a Storage Policy object
**/
export class StoragePolicyMini extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: StoragePolicyMiniTypeEnum;
}

File: sdks/go/pkg/models/shared/storagepolicymini.go
package shared

type StoragePolicyMiniTypeEnum string

const (
	StoragePolicyMiniTypeEnumStoragePolicy StoragePolicyMiniTypeEnum = "storage_policy"
)

// StoragePolicyMini
// A mini description of a Storage Policy object
type StoragePolicyMini struct {
	ID   *string                    `json:"id,omitempty"`
	Type *StoragePolicyMiniTypeEnum `json:"type,omitempty"`
}
